abort/index.d.ts

```diff
@@ -0,0 +1,30 @@
+import { Effect, Unit, UnitValue, Domain } from 'effector';
+
+type IfAny<T, Y, N> = 0 extends 1 & T ? Y : N;
+type NoAny<T> = IfAny<T, unknown, T>;
+
+export class AbortedError extends Error {
+  public aborted: true;
+  public key: string | number;
+}
+
+type MaybeAbortedError = any | AbortedError;
+
+export function abort<
+  Params,
+  Done = any,
+  Fail extends MaybeAbortedError = Error | AbortedError,
+  Signal extends Unit<string | number> = Unit<string | number>,
+>(c: {
+  domain?: Domain;
+  signal: Signal;
+  handler(
+    params: Params,
+    config: { onAbort: (callback: () => void) => () => void },
+  ): Done;
+  getKey(params: Parameters<typeof c['handler']>[0]): UnitValue<typeof c['signal']>;
```

Comment:
Could we make `getKey` optional?

In my mind, it's a popular situation, when you have to cancel all in flight effects and starts new one.

Reply:
I'm not sure about that 🤔
What kind of key should be default for that?

Maybe it is better to make `getKey` required for now and add such defaults only after real usage feedback?

Also, i was thinking of something like special values, that can be used for common cases 🤔

```ts
import {
  ALL,
  abort
} from 'patronum/abort'

cancel(ALL) // will cancel all calls in flight, no matter what kind of keys were used for each call
```

Reply:
I think, `() => ALL` (`Symbol`, maybe?) is a good default for the `getKey`.

Reply:
Should this ALL value be available, even if other getKey option is set?

Like 
```ts
abort({
 signal,
 handler,
 getKey: p => p.id // can abort by id
})

signal(ALL) // also still can abort all, by using magic value

```

Reply:
I think so.